uniffi_bindgen/src/bindings/gecko/templates/macros.cpp

```diff
@@ -0,0 +1,120 @@
+      {%- match ReturnPosition::for_type(type_) -%}
+        {%- when ReturnPosition::Return with (type_) -%} {{ type_|ret_type_cpp }}
+        {%- when ReturnPosition::OutParam with (_) -%} void
+      {%- endmatch -%}
+    {%- else -%} void
+  {%- endmatch -%}
+{%- endmacro -%}
+
+{# /* Declares a list of arguments for a WebIDL constructor. A constructor takes
+      a `GlobalObject&` as its first argument, followed by its declared
+      arguments, and then an optional `ErrorResult` if it throws. */ #}
+{%- macro decl_constructor_args(cons) -%}
+  GlobalObject& aGlobal
+  {%- let args = cons.arguments() -%}
+  {%- if !args.is_empty() -%}, {%- endif -%}
+  {%- for arg in args -%}
+  {{ arg.type_()|arg_type_cpp }} {{ arg.name() }}{%- if !loop.last %}, {%- endif -%}
+  {%- endfor -%}
+  {%- if cons.throws().is_some() -%}
+  , ErrorResult& aRv
+  {%- endif -%}
+{%- endmacro -%}
+
+{# /* Declares a list of arguments for a WebIDL static method. A static or
+      namespace method takes a `GlobalObject&` as its first argument, followed
+      by its declared arguments, an optional "out param" for the return value,
+      and an optional `ErrorResult` if it throws. */ #}
+{%- macro decl_static_method_args(cons) -%}
+  GlobalObject& aGlobal
+  {%- let args = func.arguments() %}
+  {%- if !args.is_empty() -%},{%- endif %}
+  {%- for arg in args %}
+  {{ arg.type_()|arg_type_cpp }} {{ arg.name() }}{%- if !loop.last %},{% endif %}
+  {%- endfor -%}
+  {%- call _decl_out_param(func) -%}
+  {%- if cons.throws().is_some() %}
+  , ErrorResult& aRv
+  {%- endif %}
+{%- endmacro -%}
+
+{# /* Declares a list of arguments for a WebIDL interface method. An interface
+      method takes its declared arguments, an optional "out param" for the
+      return value, and an `ErrorResult&`. */ #}
+{%- macro decl_method_args(func) -%}
+  {%- let args = func.arguments() -%}
+  {%- for arg in args -%}
+  {{ arg.type_()|arg_type_cpp }} {{ arg.name() -}}{%- if !loop.last -%},{%- endif -%}
+  {%- endfor -%}
+  {%- call _decl_out_param(func) -%}
+  {%- match func.return_type() -%}
+    {%- when Some with (type_) -%}
+      {%- match ReturnPosition::for_type(type_) -%}
+        {%- when ReturnPosition::OutParam with (type_) -%},
+        {%- else -%}
+          {%- if !args.is_empty() %}, {% endif -%}
+      {%- endmatch -%}
+    {%- else -%}
+      {%- if !args.is_empty() %}, {% endif -%}
+  {%- endmatch -%}
+  ErrorResult& aRv
+{%- endmacro -%}
+
+{# /* Returns a result from a function or method. This lifts the result from the
+      given `var`, and returns it by value or via an "out param" depending on
+      the function's return type. */ #}
+{%- macro return(func, var) -%}
+{% match func.return_type() -%}
+{%- when Some with (type_) -%}
+  {% match ReturnPosition::for_type(type_) -%}
+  {%- when ReturnPosition::OutParam with (type_) -%}
+  DebugOnly<bool> ok_ = detail::ViaFfi<{{ type_|type_cpp }}, {{ type_|ret_type_ffi }}>::Lift({{ var }}, aRetVal_);
+  MOZ_ASSERT(ok_);
```

Comment:
We should have already asserted before returning `false` in the serializers—maybe this deserves a `MOZ_RELEASE_ASSERT`?